packages/admin/cms-admin/src/generator/future/generator.ts

```diff
@@ -25,14 +25,31 @@ export type FormFieldConfig<T> = (
     | { type: "asyncSelect"; rootQuery: string; labelField?: string }
     | { type: "block"; block: ImportReference }
 ) & { name: keyof T; label?: string; required?: boolean; validate?: ImportReference; helperText?: string; readOnly?: boolean };
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+export function isFormFieldConfig<T>(arg: any): arg is FormFieldConfig<T> {
+    return arg.name !== undefined;
+}
+
+export type FormLayoutConfig<T> = {
+    type: "accordion";
+    title: string;
+    supportText?: string; // can contain field-placeholder
+    collapsible: boolean; // default true
+    initiallyExpanded: boolean; // default false
+    fields: FormFieldConfig<T>[];
+};
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+export function isFormLayoutConfig<T>(arg: any): arg is FormLayoutConfig<T> {
+    return arg.name === undefined;
```

Comment:
name == undefined is imho too unspecific

can't you use arg.type == "accordion" here? (yes, that needs other types if additional layouts are supported in future)